R/iterators.R

```diff
@@ -517,30 +517,33 @@ simple_vs_index <- function(x, i, na_ok = FALSE) {
 #' V(g)[.env$x]
 #'
 `[.igraph.vs` <- function(x, ..., na_ok = FALSE) {
-  args <- lazy_dots(..., .follow_symbols = FALSE)
+  args <- rlang::enquos(...)
 
-  ## If indexing has no argument at all, then we still get one,
-  ## but it is "empty", a name that is  ""
+  ## Special case, no argument
+  if (length(args) == 0) {
+    return(x)
+  }
 
-  ## Special case, no argument (but we might get an artificial
-  ## empty one
-  if (length(args) < 1 ||
-    (length(args) == 1 && inherits(args[[1]]$expr, "name") &&
-      as.character(args[[1]]$expr) == "")) {
+  ## or empty (what is the test case for this??)
+  ## try() because of .env$something would otherwise error
+  first_is_empty <- !nzchar(paste(rlang::quo_get_expr(args[[1]]), collapse = ""))
```

Comment:
`rlang::is_missing(args[[1]])` ? But we need a test case...

The `.ignore_empty` argument to `rlang::enquos()` might take care of that.